src/wazuh_db/wdb_syscollector.c

```diff
@@ -833,6 +851,10 @@ int wdb_port_save(wdb_t * wdb, const char * scan_id, const char * scan_time, con
 int wdb_port_insert(wdb_t * wdb, const char * scan_id, const char * scan_time, const char * protocol, const char * local_ip, int local_port, const char * remote_ip, int remote_port, int tx_queue, int rx_queue, long long inode, const char * state, int pid, const char * process, const char * checksum, const char * item_id, const bool replace) {
     sqlite3_stmt *stmt = NULL;
 
+    if (!local_ip) {
```

Comment:
Just wondering why not checking protocol ?
`PRIMARY KEY (protocol, local_ip, local_port, inode)`

Reply:
My bad, added protocol and also check those int values because when they are < 0 they bind_null.